src/WebJobs.Script.WebHost/ContainerManagement/LinuxFunctionExecutionActivityPublisher.cs

```diff
@@ -0,0 +1,174 @@
+            {
+                throw new NotSupportedException(
+                    $"{nameof(LinuxFunctionExecutionActivityPublisher)} is available in Linux consumption environment only");
+            }
+
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+            _meshInitServiceClient = meshInitServiceClient;
+            _logger = logger;
+            _flushIntervalMs = flushIntervalMs;
+            _timer = new Timer(OnTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _queue = new ConcurrentQueue<ContainerFunctionExecutionActivity>();
+            _flushInProgress = 0;
+        }
+
+        private void Start()
+        {
+            _logger.LogInformation($"Starting {nameof(LinuxFunctionExecutionActivityPublisher)}");
+
+            // start the timer by setting the due time
+            SetTimerInterval(_flushIntervalMs);
+        }
+
+        private void OnStandbyOptionsChange()
+        {
+            _logger.LogInformation($"Triggering {nameof(OnStandbyOptionsChange)}");
+
+            if (!_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                Start();
+            }
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Initializing {nameof(LinuxFunctionExecutionActivityPublisher)}");
+
+            if (_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _logger.LogInformation($"Registering {nameof(_standbyOptionsOnChangeSubscription)}");
+                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange());
+            }
+            else
+            {
+                Start();
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Stopping {nameof(LinuxFunctionExecutionActivityPublisher)}");
+
+            // stop the timer if it has been started
+            _timer?.Change(Timeout.Infinite, Timeout.Infinite);
+
+            return Task.CompletedTask;
+        }
+
+        private async void OnTimer(object state)
+        {
+            await FlushFunctionExecutionActivities();
+            SetTimerInterval(_flushIntervalMs);
+        }
+
+        private async Task FlushFunctionExecutionActivities()
+        {
+            try
+            {
+                if (Interlocked.CompareExchange(ref _flushInProgress, 1, 0) == 0)
+                {
+                    try
+                    {
+                        var uniqueActivities = new HashSet<ContainerFunctionExecutionActivity>();
+
+                        while (_queue.TryDequeue(out var a))
+                        {
+                            uniqueActivities.Add(a);
+                        }
+
+                        if (uniqueActivities.Any() || _lastHeartBeatTime.AddMinutes(5) < DateTime.UtcNow)
```

Comment:
I don't understand what you're doing with this heartbeat and why you have 2 separate uniqueActivities.Any() blocks?

Reply:
Simplified the logic a bit. Heartbeat is just so we know the service is running.